Develop1

SuvorovaTatyanaSergeevna opened the pull request:
Написаны юнит-тесты для программы по заказу бургеров из первой части диплома.
Добавлены библиотеки JUnit4, Mockito и плагин Jacoco.
Добавлен отчет с процентным покрытием тестов.

SuvorovaTatyanaSergeevna committed:
Подключены плагин Jacoco, библиотеки Junit 4 и Mockito.

SuvorovaTatyanaSergeevna committed:
Создан тестовый класс для Bun с моками и стабами.

SuvorovaTatyanaSergeevna committed:
Создан тестовый класс для класса Bun с параметризацией.

SuvorovaTatyanaSergeevna committed:
Создан тестовый класс для класса Burger с моками и стабами.

SuvorovaTatyanaSergeevna committed:
Создан тестовый класс для класса Ingredient с моками и стабами.

SuvorovaTatyanaSergeevna committed:
Создан тестовый класс для класса Ingredient с параметризацией.

SuvorovaTatyanaSergeevna committed:
Создан тестовый класс для класса c базой данных Database.

SuvorovaTatyanaSergeevna committed:
Отформатирован код

SuvorovaTatyanaSergeevna committed:
Добавила папку с отчетом Jacoco

mkouznetsov-git commented on pom.xml:
```diff
@@ -12,5 +12,44 @@
         <maven.compiler.source>8</maven.compiler.source>
```
Версия джавы должна быть 11

mkouznetsov-git commented on src/test/java/BunParameterizedTest.java:
```diff
@@ -0,0 +1,38 @@
+    @Parameterized.Parameters
+    public static Object[][] getBunParameters() {
+        return new Object[][]{
+                {"", 50.0f},
```
Здесь надо выделить хотя бы базовые "классы эквивалентности" для проверки булки. Для названий -- пустая строка (у тебя есть), null, слишком длинная строка, спецсимволы в строке и т.п. Для цены -- отрицательная цена, ноль, минимально возможная положительная (дробная) цена, максимально возможная положительная цена, и т.п.

SuvorovaTatyanaSergeevna committed:
Исправлена версия Maven.compiler.
Добавлены новые ингредиенты и цены в BunParametrizedTest.

SuvorovaTatyanaSergeevna committed:
Увеличение покрытия.
Изменила тестовый метод getReceiptBurgerTest, чтобы увеличить покрытие тестами проекта. Сформирован новый отчет Jacoco.

SuvorovaTatyanaSergeevna committed:
Оформила файл README.md
